// open-korean-text-node/lib/processor/tokens.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const classes_1 = require("../classes");
const processor_1 = require("./processor");
class IntermediaryTokensObject extends classes_1.Seq {
    toJSON() {
        return this.map((item) => {
            const token = {
                text: item.text(),
                pos: item.pos().toString(),
                offset: item.offset(),
                length: item.length(),
                isUnknown: item.unknown()
            };
            if (item.stem().nonEmpty()) {
                token.stem = item.stem().get();
            }
            return token;
        });
    }
    extractPhrases(options) {
        return processor_1.OpenKoreanTextProcessor.extractPhrases(this, options);
    }
    extractPhrasesSync(options) {
        return processor_1.OpenKoreanTextProcessor.extractPhrasesSync(this, options);
    }
    detokenize() {
        return processor_1.OpenKoreanTextProcessor.detokenize(this);
    }
    detokenizeSync() {
        return processor_1.OpenKoreanTextProcessor.detokenizeSync(this);
    }
}
exports.IntermediaryTokensObject = IntermediaryTokensObject;
