// open-korean-text-node/lib/classes/collections.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const abstract_1 = require("./abstract");
class Collections extends abstract_1.AbstractJavaClass {
    map(mapFunc) {
        let index = 0;
        const it = this.interface.iterator();
        const res = [];
        while (it.hasNext()) {
            res.push(mapFunc(it.next(), index++));
        }
        return res;
    }
}
Collections.className = 'java.util.Collections';
exports.Collections = Collections;
class ArrayList extends Collections {
    constructor(values) {
        super();
        if (values && values.length) {
            this.addAllSync(values);
        }
    }
    add(item) {
        return this.interface.addPromise(item);
    }
    addSync(item) {
        return this.interface.add(item);
    }
    addAll(items) {
        return Promise.all(items.map(item => this.add(item)));
    }
    addAllSync(items) {
        items.forEach(item => this.addSync(item));
    }
}
ArrayList.className = 'java.util.ArrayList';
exports.ArrayList = ArrayList;
class Seq extends Collections {
}
Seq.className = 'scala.collection.Seq';
exports.Seq = Seq;
