// open-korean-text-node/lib/classes/tokens.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const abstract_1 = require("./abstract");
class KoreanTokenObject extends abstract_1.AbstractJavaClass {
    constructor(text, pos, offset, length, isUnknown, stem) {
        super(text, KoreanPosObject.valueOf(pos), offset, length, !!isUnknown, stem);
        this.text = text;
        this.pos = pos;
        this.offset = offset;
        this.length = length;
        this.isUnknown = isUnknown;
        this.stem = stem;
    }
    toJSON() {
        return {
            text: this.text,
            stem: this.stem,
            pos: this.pos,
            offset: this.offset,
            length: this.length,
            isUnknown: this.isUnknown,
        };
    }
}
KoreanTokenObject.className = 'org.openkoreantext.processor.KoreanTokenJava';
exports.KoreanTokenObject = KoreanTokenObject;
class KoreanPosObject extends abstract_1.AbstractJavaClass {
    static valueOf(name) {
        return this.class.valueOf(name);
    }
    static get Noun() { return this.class.Noun; }
    static get Verb() { return this.class.Verb; }
    static get Adjective() { return this.class.Adjective; }
    static get Adverb() { return this.class.Adverb; }
    static get Determiner() { return this.class.Determiner; }
    static get Exclamation() { return this.class.Exclamation; }
    static get Josa() { return this.class.Josa; }
    static get Eomi() { return this.class.Eomi; }
    static get PreEomi() { return this.class.PreEomi; }
    static get Conjunction() { return this.class.Conjunction; }
    static get NounPrefix() { return this.class.NounPrefix; }
    static get VerbPrefix() { return this.class.VerbPrefix; }
    static get Suffix() { return this.class.Suffix; }
    static get Unknown() { return this.class.Unknown; }
    static get Korean() { return this.class.Korean; }
    static get Foreign() { return this.class.Foreign; }
    static get Number() { return this.class.Number; }
    static get KoreanParticle() { return this.class.KoreanParticle; }
    static get Alpha() { return this.class.Alpha; }
    static get Punctuation() { return this.class.Punctuation; }
    static get Hashtag() { return this.class.Hashtag; }
    static get ScreenName() { return this.class.ScreenName; }
    static get Email() { return this.class.Email; }
    static get URL() { return this.class.URL; }
    static get CashTag() { return this.class.CashTag; }
    static get Space() { return this.class.Space; }
    static get Others() { return this.class.Others; }
    static get ProperNoun() { return this.class.ProperNoun; }
}
KoreanPosObject.className = 'org.openkoreantext.processor.KoreanPosJava';
exports.KoreanPosObject = KoreanPosObject;
